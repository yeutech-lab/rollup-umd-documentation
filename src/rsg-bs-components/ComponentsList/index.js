export { default } from 'rsg-components/ComponentsList/ComponentsList';
