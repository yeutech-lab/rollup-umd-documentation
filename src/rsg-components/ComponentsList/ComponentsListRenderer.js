import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Link from 'rsg-components/Link';
import Styled from 'rsg-components/Styled';

const styles = ({ color, fontFamily, fontSize, space, mq }) => ({
	list: {
		margin: 0,
		paddingLeft: space[2],
	},
	item: {
		color: color.base,
		display: 'block',
		margin: [[space[1], 0, space[1], 0]],
		fontFamily: fontFamily.base,
		fontSize: fontSize.base,
		listStyle: 'none',
		overflow: 'hidden',
		textOverflow: 'ellipsis',
	},
	isChild: {
		[mq.small]: {
			display: 'inline-block',
			margin: [[0, space[1], 0, 0]],
		},
	},
	heading: {
		color: color.base,
		marginTop: space[1],
		fontFamily: fontFamily.base,
		fontWeight: 'bold',
	},
});

export function ComponentsListRenderer({ classes, items }) {
	items = items.filter(item => item.name);

	if (!items.length) {
		return null;
	}

	return (
		<ul className={classes.list}>
			{items.map(({ heading, name, href, content }) => (
				<li
					className={cx(classes.item, (!content || !content.props.items.length) && classes.isChild)}
					key={href}
				>
					<Link className={cx(heading && classes.heading)} href={href}>
						{name}
					</Link>
					{content}
				</li>
			))}
		</ul>
	);
}

ComponentsListRenderer.propTypes = {
	items: PropTypes.array.isRequired,
	classes: PropTypes.object.isRequired,
	useIsolatedLinks: PropTypes.bool,
};

export default Styled(styles)(ComponentsListRenderer);
