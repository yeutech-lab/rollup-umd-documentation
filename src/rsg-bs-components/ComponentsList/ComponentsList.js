import React from 'react';
import PropTypes from 'prop-types';
import getUrl from 'react-styleguidist/lib/utils/getUrl';
import ComponentsListRenderer from './ComponentsListRenderer';

function ComponentsList({ items, useIsolatedLinks = false, isOpenCollapse }) {
  const mappedItems = items.map((item) => ({
    ...item,
    href: getUrl({
      name: item.name,
      slug: item.slug,
      anchor: !useIsolatedLinks,
      isolated: useIsolatedLinks,
    }),
  }));
  return <ComponentsListRenderer items={mappedItems} isOpenCollapse={isOpenCollapse} />;
}

ComponentsList.propTypes = {
  items: PropTypes.array.isRequired,
  useIsolatedLinks: PropTypes.bool, // eslint-disable-line react/require-default-props
  isOpenCollapse: PropTypes.bool.isRequired,
};

export default ComponentsList;

