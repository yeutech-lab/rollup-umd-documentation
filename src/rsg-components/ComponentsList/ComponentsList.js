import React from 'react';
import ComponentsListRenderer from 'rsg-components/ComponentsList/ComponentsListRenderer';
import PropTypes from 'prop-types';
import getUrl from '../../utils/getUrl';

function ComponentsList({ classes, items, useIsolatedLinks = false }) {
	const mappedItems = items.map(item => ({
		...item,
		href: getUrl({
			name: item.name,
			slug: item.slug,
			anchor: !useIsolatedLinks,
			isolated: useIsolatedLinks,
		}),
	}));
	return <ComponentsListRenderer classes={classes} items={mappedItems} />;
}

ComponentsList.propTypes = {
	items: PropTypes.array.isRequired,
	classes: PropTypes.object,
	useIsolatedLinks: PropTypes.bool,
};

export default ComponentsList;
