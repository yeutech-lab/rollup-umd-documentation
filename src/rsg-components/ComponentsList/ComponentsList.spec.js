import React from 'react';
import ComponentsList from './ComponentsList';
import { ComponentsListRenderer } from './ComponentsListRenderer';

it('should set the correct href for items', () => {
	const components = [
		{
			name: 'Button',
			slug: 'button',
		},
		{
			name: 'Input',
			slug: 'input',
		},
	];

	const actual = shallow(<ComponentsList items={components} classes={{}} />);
	expect(actual).toMatchSnapshot();
});

it('should set the correct href for items when isolated links should be used', () => {
	const components = [
		{
			name: 'Button',
			slug: 'button',
		},
		{
			name: 'Input',
			slug: 'input',
		},
	];

	const actual = shallow(<ComponentsList items={components} classes={{}} useIsolatedLinks />);
	expect(actual).toMatchSnapshot();
});

it('should render sections with nested components', () => {
	const components = [
		{
			name: 'Button',
			slug: 'button',
			href: '#button',
		},
		{
			name: 'Input',
			slug: 'input',
			href: '#input',
		},
	];
	const actual = shallow(<ComponentsListRenderer items={components} classes={{}} />);

	expect(actual).toMatchSnapshot();
});

it('should return null when the list is empty', () => {
	const actual = shallow(<ComponentsListRenderer items={[]} classes={{}} />);

	expect(actual.getElement()).toBe(null);
});

it('should ignore items without name', () => {
	const components = [
		{
			name: 'Button',
			slug: 'button',
			href: '#button',
		},
		{
			slug: 'input',
			href: '#input',
		},
	];
	const actual = shallow(<ComponentsListRenderer items={components} classes={{}} />);

	expect(actual).toMatchSnapshot();
});
