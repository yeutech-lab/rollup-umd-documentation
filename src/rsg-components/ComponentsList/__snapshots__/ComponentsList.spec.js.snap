// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should ignore items without name 1`] = `
<ul>
  <li
    className=""
    key="#button"
  >
    <Styled(Link)
      className=""
      href="#button"
    >
      Button
    </Styled(Link)>
  </li>
</ul>
`;

exports[`should render sections with nested components 1`] = `
<ul>
  <li
    className=""
    key="#button"
  >
    <Styled(Link)
      className=""
      href="#button"
    >
      Button
    </Styled(Link)>
  </li>
  <li
    className=""
    key="#input"
  >
    <Styled(Link)
      className=""
      href="#input"
    >
      Input
    </Styled(Link)>
  </li>
</ul>
`;

exports[`should set the correct href for items 1`] = `
<Styled(ComponentsList)
  classes={Object {}}
  items={
    Array [
      Object {
        "href": "blank#button",
        "name": "Button",
        "slug": "button",
      },
      Object {
        "href": "blank#input",
        "name": "Input",
        "slug": "input",
      },
    ]
  }
/>
`;

exports[`should set the correct href for items when isolated links should be used 1`] = `
<Styled(ComponentsList)
  classes={Object {}}
  items={
    Array [
      Object {
        "href": "blank#!/Button",
        "name": "Button",
        "slug": "button",
      },
      Object {
        "href": "blank#!/Input",
        "name": "Input",
        "slug": "input",
      },
    ]
  }
/>
`;
